tests/functional/test_install.py

```diff
@@ -486,6 +487,69 @@ def test_install_package_with_root(script, data):
     assert root_path in result.files_created, str(result)
 
 
+def test_install_package_with_prefix(script, data):
+    """
+    Test installing a package using pip install --prefix
+    """
+    prefix_path = script.scratch_path / 'prefix'
+    result = script.pip(
+        'install', '--prefix', prefix_path, '-f', data.find_links,
+        '--no-binary', 'simple', '--no-index', 'simple==1.0',
+    )
+
+    if hasattr(sys, "pypy_version_info"):
+        path = script.scratch / 'prefix'
+    else:
+        path = script.scratch / 'prefix' / 'lib' / 'python{0}'.format(pyversion)  # noqa
+    install_path = (
+        path / 'site-packages' / 'simple-1.0-py{0}.egg-info'.format(pyversion)
+    )
+    assert install_path in result.files_created, str(result)
+
+
+def test_install_editable_with_prefix(script):
+    # make a dummy project
+    pkga_path = script.scratch_path / 'pkga'
+    pkga_path.mkdir()
+    pkga_path.join("setup.py").write(textwrap.dedent("""
+        from setuptools import setup
+        setup(name='pkga',
+              version='0.1')
+    """))
+
+    site_packages = os.path.join(
+        'prefix', 'lib', 'python{0}'.format(pyversion), 'site-packages')
+
+    # make sure target path is in PYTHONPATH
+    pythonpath = script.scratch_path / site_packages
+    pythonpath.makedirs()
+    script.environ["PYTHONPATH"] = pythonpath
```

Comment:
Indeed, for the curious, you get something like:

        running develop
        Checking .pth file support in /home/xfernandez/tmp/lib/python2.7/site-packages
        /home/xfernandez/.virtualenvs/c13ba271efca52d0/bin/python -E -c pass
        TEST FAILED: /home/xfernandez/tmp/lib/python2.7/site-packages does NOT support .pth files
        error: bad install directory or PYTHONPATH
        
        You are attempting to install a package to a directory that is not
        on PYTHONPATH and which Python does not read ".pth" files from.  The
        installation directory you specified (via --install-dir, --prefix, or
        the distutils default setting) was:
        
            /home/xfernandez/tmp/lib/python2.7/site-packages
        
        and your PYTHONPATH environment variable currently contains:
        
            ''
        
        Here are some of your options for correcting the problem:
        
        * You can choose a different installation directory, i.e., one that is
          on PYTHONPATH or supports .pth files
        
        * You can add the installation directory to the PYTHONPATH environment
          variable.  (It must then also be on PYTHONPATH whenever you run
          Python and want to use the package(s) you are installing.)
        
        * You can set up the installation directory to support ".pth" files by
          using one of the approaches described here:
        
          https://pythonhosted.org/setuptools/easy_install.html#custom-installation-locations
        
        Please make the appropriate changes for your system and try again.
        
        ----------------------------------------
    Command "/home/xfernandez/.virtualenvs/c13ba271efca52d0/bin/python -c "import setuptools,
    tokenize; __file__='/home/xfernandez/dev/blease/setup.py'; exec(compile(getattr(tokenize, 'open',
    open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps
    --prefix=/home/xfernandez/tmp" failed with error code 1 in /home/xfernandez/dev/blease